zeeguu/core/word_scheduling/basicSR/basicSR.py

```diff
@@ -52,29 +58,36 @@ def update_schedule(self, db_session, correctness):
                 db.session.commit()
                 return
 
-            if datetime.now() < self.next_practice_time:
+            # Use the same logic as when selecting bookmarks
+            # Avoid case where if schedule at 01-01-2024 11:00 and user does it at
+            # 01-01-2024 10:00 the status is not updated.
+            if self.get_current_study_window() < self.next_practice_time:
                 # a user might have arrived here by doing the
                 # bookmarks in a text for a second time...
                 # in general, as long as they didn't wait for the
                 # cooldown perio, they might have arrived to do
                 # the exercise again; but it should not count
                 return
-
-            new_cooling_interval = NEXT_COOLING_INTERVAL_ON_SUCCESS[
-                self.cooling_interval
-            ]
-            next_practice_date = datetime.now() + timedelta(
-                minutes=new_cooling_interval
+            # Since we can now loose the streak on day 8,
+            # we might have to repeat it a few times to learn it.
+            new_cooling_interval = NEXT_COOLING_INTERVAL_ON_SUCCESS.get(
+                self.cooling_interval, 8 * ONE_DAY
             )
-            self.cooling_interval = new_cooling_interval
-            self.next_practice_time = next_practice_date
             self.consecutive_correct_answers += 1
```

Comment:
i guess with the leitner system we don't even need the consecutive_correct_answers?

Reply:
True, however we need to make sure we update the strings to reflect that we do not care about the consecutive number of answers.